blockchain/types/tx_internal_data.go

```diff
@@ -610,14 +608,7 @@ func IntrinsicGas(data []byte, accessList AccessList, contractCreation bool, r p
 		gas = params.TxGas
 	}
 
-	var gasPayloadWithGas uint64
-	var err error
-	if r.IsIstanbul {
-		gasPayloadWithGas, err = IntrinsicGasPayload(gas, data, contractCreation, r)
-	} else {
-		gasPayloadWithGas, err = IntrinsicGasPayloadLegacy(gas, data)
-	}
-
+	gasPayloadWithGas, err := IntrinsicGasPayload(gas, data, contractCreation, rules)
```

Comment:
Can we make it more clearer?

```suggestion
	payloadGas, err := IntrinsicGasPayload(data, isContractCreation, rules)
	if err != nil {
		return 0, err
	}
	gas += payloadGas
```

Reply:
Some tx types have predefined gas values, so as I understand it, `IntrinsicGasPayload` is used to retain the gas value at its call site.

Reply:
Because the IntrinsicGas() function is calculating a sum of three components,
```
IntrinsicGas = (TxGasContractCreation or TxGas) + PayloadGas + (AcesssLisstGas)
```
I thought we can make this function look like roughly
```
gas = TxGasContractCreation or TxGas
gas += PayloadGas
gas += AccessListGas
```
But for the uint64 overflow checks, now I see that it's inevitable to pass in the `gas` value into `IntrinsicGasPayload` function.